### ISSUE TITLE
Encode or hide sensitive data in the request url 

### ISSUE THREAD

**Issue by RomChig** [wannesvr/dota2api#13]
Hi @wannesvr I'd suggest hiding apiKey or encode them in logs, let me know your thoughts reg that plz

**Comment by wannesvr** [comment]
Hi @RomChig,

Fully agree. I was also thinking of making the logging debug-only or just removing it completely.
I don't think it provides that much useful insight into what's happening, and any errors should result in exceptions anyway.

What do you think?

**Comment by RomChig** [comment]
@wannesvr I'd prefer here to keep logging catastrophic situations (e.g. incorrect apiKey or some problems with an attempt to send a request connection etc it should be present to let clients of the lib minimize lookup when exception occurred ). It's a good idea reg debug mod, in this case.

**Comment by wannesvr** [comment]
Wouldn’t it be preferable to throw an exception anytime something critical goes wrong? In fact it already throws exceptions when it can’t reach Steam API or the request fails (status code != 200), and soon will throw a proper exception when using a wrong API key

I feel like the actual logging of errors should be the responsibility of the application using the library.

**Comment by RomChig** [comment]
Yes, fail-fast is a best practice, I mean that besides throwing an exception we could log it with an appropriate message and set a cause or issue in a constructor as well.
Example:
```
public Dota2ApiException(String message, Throwable cause) {
	super(message, cause);
}
```
and somewhere(usually in catch blocks):
log.error("Invalid API key", "someParams if needed here");
throw new Dota2ApiException("Invalid API key", e.getCause());

**Comment by wannesvr** [comment]
Yes I was planning on passing the cause in the Dota2ApiException in #9. I'm not sure I see the value of the logging since the same message will (probably) be passed in the exception that gets thrown, as shown in your example.

I'll add the logging so you can at least see the url the library accesses, with the api key hidden.

**Comment by wannesvr** [comment]
Fixed in #14
